micro-rdk-server/native/main.rs

```diff
@@ -83,7 +83,7 @@ mod native {
             mdns,
             Box::new(network),
         );
-        server.run_forever();
+        server.run_forever::<bytes::BytesMut>();
```

Comment:
I don't quite see a reason to?

Reply:
just that the leap from `BytesMut` to `NativeHeapAllocation` feels odd. One is external with a generic name, not hinting on how it might work underneath, the other is the opposite (granted, because we're the ones implementing it)

Reply:
Ok, that's a fair point about clarity, I'll make this change